### ISSUE TITLE
Switch to use Chrome v59 Headless browser

### ISSUE THREAD

**Issue by gajus** [pocketjoso/penthouse#169]
There are a few simple abstractions, e.g. https://github.com/gwuhaolin/chrome-render

**Comment by pocketjoso** [comment]
Yep I've been thinking about it, but not sure I will be able to find the time for it. I would be open for a PR for it, although I understand that it would be a big code change.

**Comment by gajus** [comment]
> although I understand that it would be a big code change.

I am not familiar with penthouse code base, but as far as I can think, all you need to know about CDP to achieve this is:

```js
const chromeLauncher = require('chrome-launcher');
const CDP = require('chrome-remote-interface');

function launchChrome(headless=true) {
  return chromeLauncher.launch({
    chromeFlags: [
      '--window-size=412,732',
      '--disable-gpu',
      headless ? '--headless' : ''
    ]
  });
}

launchChrome().then(async chrome => {
  const protocol = await CDP({
    port: chrome.port
  });

  const {
    Page,
    Runtime,
    CSS,
    DOM
  } = protocol;

  await Page.enable();
  await DOM.enable();
  await CSS.enable();

  Page.navigate({
    url: 'https://go2cinema.com/movies/baywatch-2017-1198354'
  });
  
  Page.loadEventFired(async () => {
    // Run whatevern code in the page.
    const result = await Runtime.evaluate({
      expression: `document.querySelector('title').textContent`
    });

    const rootNode = await DOM.getDocument();

    // Query inidividual document elements
    const node = await DOM.querySelector({
      nodeId: rootNode.root.nodeId,
      selector: 'html'
    });

    // Get CSS class names used in the document
    const result = await DOM.collectClassNamesFromSubtree({
      nodeId: node.nodeId
    });
  });
});

```

**Comment by gajus** [comment]
Solved by https://github.com/gajus/usus

**Comment by pocketjoso** [comment]
@gajus Cool to see your library, congrats on the launch.

I will still be looking to migrate Penthouse to headless chrome to not stagnate with Penthouse, but I will keep the code closer to what I have already, so it will be interesting to compare the Penthouse and usus afterward. I'm all for open source competition - let users use the best library! :)

**Comment by gajus** [comment]
Or you could just contribute to usus instead of recreating the same functionality.

Anyway, you don't need to deal with the low level API. You can use https://github.com/GoogleChrome/puppeteer

**Comment by pocketjoso** [comment]
@gajus - I would not want to just drop all the people who depend on Penthouse currently, and besides I do see it as an interesting thing to work on myself. I have more time available right now too then when you originally opened the repo.

I did already check out `puppeteer` last night and it seems like it would be very quick to migrate from phantomjs there as the api's are quite similar.

**Comment by pocketjoso** [comment]
This is almost done now, in #174.

**Comment by pocketjoso** [comment]
Resolved by #174
